src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -52,6 +52,7 @@ public function __construct()
             new ResolveReferencesToAliasesPass(),
             new ResolveInvalidReferencesPass(),
             new AnalyzeServiceReferencesPass(true),
+            new AutowiringPass(),
```

Comment:
this must be before the ``AnalyzeServiceReferencesPass``, otherwise the changes done by the autowiring pass won't be taken into account when checking for circular references